src/app_service/service/node/service.nim

```diff
@@ -0,0 +1,118 @@
+import NimQml, chronicles, strutils, json, nimcrypto
+import status/statusgo_backend_new/core as status_core
+import status/utils
+import eventemitter
+
+import ../settings/service as settings_service
+
+import ../../../app/core/tasks/[qt, threadpool]
+
+include async_tasks
+
+logScope:
+  topics = "node-service"
+
+# Signals which may be emitted by this service:
+const SIGNAL_BITS_SET_FETCHED* = "SIGNAL_BITS_SET_FETCHED" 
+
+type
+    BitsSet* = ref object of Args
+        bitsSet*: int
+
+QtObject:
+    type Service* = ref object of QObject
+        events*: EventEmitter
+        threadpool: ThreadPool
+        settingsService: settings_service.Service
+        bloomBitsSet: int
+        peers*: seq[string]
+        connected: bool
+
+    proc delete*(self: Service) =
+       self.QObject.delete
+
+    proc newService*(events: EventEmitter, threadpool: ThreadPool, settingsService: settings_service.Service): Service =
+        new(result, delete)
+        result.QObject.setup
+        result.events = events
+        result.threadpool = threadpool
+        result.settingsService = settingsService
+        result.peers = @[]
+        result.connected = false
+
+    proc init*(self: Service) =
+        discard
+
+    proc bloomFiltersBitsSet*(self: Service, slot: string) =
+        let arg = BloomBitsSetTaskArg(
+            tptr: cast[ByteAddress](bloomBitsSetTask),
+            vptr: cast[ByteAddress](self.vptr),
+            slot: slot
+        )
+        self.threadpool.start(arg)
+
+    proc fetchBitsSet*(self: Service) =
+        self.bloomFiltersBitsSet("bitsSet")
+
+    proc getBloomBitsSet*(self: Service): int =
+        return self.bloomBitsSet;
+
+    proc bitsSet*(self: Service, bitsSet: string) {.slot.} =
+        self.bloomBitsSet = parseInt(bitsSet)
+        self.events.emit(SIGNAL_BITS_SET_FETCHED, BitsSet(bitsSet: self.bloomBitsSet))
+
+    proc sendRPCMessageRaw*(self: Service, msg: string): string =
+        echo "sending RPC message"
+        status_core.callPrivateRPCRaw(msg)
+
+    proc adminPeers*(): seq[string] =
+        let response = status_core.callPrivateRPC("admin_peers", %* []).result
```

Comment:
@borismelnik calls like this should go to `status-lib`, so move them there. Also take a look how it's done in other services.